receiver/awscontainerinsightreceiver/internal/stores/localnode.go

```diff
@@ -0,0 +1,184 @@
+)
+
+type LocalNodeDecorator struct {
+	hostInfo              hostInfo
+	version               string
+	nodeName              string
+	containerOrchestrator string
+	ecsInfo               EcsInfo
+	logger                *zap.Logger
+	k8sDecorator          Decorator
+}
+
+type hostInfo interface {
+	GetClusterName() string
+	GetEBSVolumeID(string) string
+	ExtractEbsIDsUsedByKubernetes() map[string]string
+	GetInstanceID() string
+	GetInstanceType() string
+	GetAutoScalingGroupName() string
+}
+
+type EcsInfo interface {
+	GetContainerInstanceID() string
+	GetClusterName() string
+}
+
+type Decorator interface {
+	Decorate(CIMetric) CIMetric
+	Shutdown() error
+}
+
+func NewLocalNodeDecorator(logger *zap.Logger, containerOrchestrator string, hostInfo hostInfo, options ...Option) (*LocalNodeDecorator, error) {
+	nodeName := os.Getenv(ci.HostName)
+	if nodeName == "" && containerOrchestrator == ci.EKS {
+		return nil, fmt.Errorf("missing environment variable %s. Please check your deployment YAML config", ci.HostName)
+	}
+
+	d := &LocalNodeDecorator{
+		hostInfo:              hostInfo,
+		version:               "0",
+		nodeName:              nodeName,
+		containerOrchestrator: containerOrchestrator,
+		logger:                logger,
+	}
+
+	for _, option := range options {
+		option(d)
+	}
+
+	return d, nil
+}
+
+type Option func(decorator *LocalNodeDecorator)
+
+func WithK8sDecorator(d Decorator) Option {
+	return func(ld *LocalNodeDecorator) {
+		ld.k8sDecorator = d
+	}
+}
+
+func WithECSInfo(f EcsInfo) Option {
+	return func(c *LocalNodeDecorator) {
+		c.ecsInfo = f
+	}
+}
+
+func (d *LocalNodeDecorator) Decorate(m CIMetric) CIMetric {
+	result := m
+
+	ebsVolumeIDsUsedAsPV := d.hostInfo.ExtractEbsIDsUsedByKubernetes()
+	tags := m.GetTags()
+	d.addEbsVolumeInfo(tags, ebsVolumeIDsUsedAsPV)
+
+	// add version
+	tags[ci.Version] = d.version
+
+	// add NodeName for node, pod and container
+	metricType := tags[ci.MetricType]
+	if d.nodeName != "" && (ci.IsNode(metricType) || ci.IsInstance(metricType) ||
+		ci.IsPod(metricType) || ci.IsContainer(metricType)) {
+		tags[ci.NodeNameKey] = d.nodeName
+	}
+
+	// add instance id and type
+	if instanceID := d.hostInfo.GetInstanceID(); instanceID != "" {
+		tags[ci.InstanceID] = instanceID
+	}
+	if instanceType := d.hostInfo.GetInstanceType(); instanceType != "" {
+		tags[ci.InstanceType] = instanceType
+	}
+
+	// add scaling group name
+	tags[ci.AutoScalingGroupNameKey] = d.hostInfo.GetAutoScalingGroupName()
+
+	// add ECS cluster name and container instance id
+	if d.containerOrchestrator == ci.ECS {
+		if d.ecsInfo.GetClusterName() == "" {
+			d.logger.Warn("Can't get cluster name")
+		} else {
+			tags[ci.ClusterNameKey] = d.ecsInfo.GetClusterName()
+		}
+
+		if d.ecsInfo.GetContainerInstanceID() == "" {
+			d.logger.Warn("Can't get containerInstanceId")
+		} else {
+			tags[ci.ContainerInstanceIDKey] = d.ecsInfo.GetContainerInstanceID()
+		}
```

Comment:
Why cant we move lines 110-120 to addECSResources func as well

Reply:
Sure, done.  I had moved this from cadvisor_linux and didn't make any modifications.